src/App.tsx

```diff
@@ -49,11 +53,15 @@ function App() {
       } finally {
         setLoading(false);
       }
-
-      abortControllerRef.current = abortController;
     };
 
     fetchData();
+
+    return () => {
+      if (abortControllerRef.current) {
+        abortControllerRef.current.abort("Component unmounted");
```

Comment:
As we discussed during the meeting, cleanup function runs not only on unmounting of a component, but also between rerenders. So, strictly speaking, "Component unmounted" is not always correct here. Just something to keep in mind.